src/pudl/io_managers.py

```diff
@@ -349,13 +356,26 @@ def load_input(self, context: InputContext) -> pd.DataFrame:
                 name.
         """
         table_name = self._get_table_name(context)
+        pkg = Package.from_resource_ids()
+        all_resources = [resource.name for resource in pkg.resources]
+        res = None
+        if table_name in all_resources:
+            res = pkg.get_resource(table_name)
+
         _ = self._get_sqlalchemy_table(table_name)
 
         engine = self.engine
 
         with engine.connect() as con:
             try:
-                df = pd.read_sql_table(table_name, con)
+                dfs = []
+                for chunk_df in pd.read_sql_table(table_name, con, chunksize=100_000):
+                    if res:
+                        chunk_df = res.enforce_schema(chunk_df)
+                    else:
+                        chunk_df = pudl.metadata.fields.apply_pudl_dtypes(chunk_df)
+                    dfs.append(chunk_df)
```

Comment:
If we decide load_input should fail if the table_name doesn't show up in our metadata, this call to `apply_pudl_dtypes` would not be required.